src/Symfony/Component/HttpKernel/CacheClearer/ChainCacheClearer.php

```diff
@@ -47,9 +47,13 @@ public function clear($cacheDir)
      * Adds a cache clearer to the aggregate.
      *
      * @param CacheClearerInterface $clearer
+     *
+     * @deprecated since version 3.4, to be removed in 4.0, inject the list of clearers as a constructor argument instead.
      */
     public function add(CacheClearerInterface $clearer)
     {
+        @trigger_error(sprintf('The "%s()" method is deprecated since version 3.4 and will be removed in 4.0, inject the list of clearers as a constructor argument instead.', __METHOD__), E_USER_DEPRECATED);
+
         $this->clearers[] = $clearer;
```

Comment:
isn't`$this->clearers` an iterator now?

Reply:
Is it, but here is my reasoning: this method is deprecated, and no code exists yet that uses both the method and an iterator. No new code should be written that would use both an iterator, and the deprecated method. Thus, all is fine :)